common/replays.ts

```diff
@@ -0,0 +1,79 @@
+import { ReplayRace } from 'jssuh'
+import { assertUnreachable } from './assert-unreachable'
+import { RaceChar } from './races'
+import { SbUserId } from './users/sb-user'
+
+export function replayRaceToChar(name: ReplayRace): RaceChar {
+  switch (name) {
+    case 'protoss':
+      return 'p'
+    case 'unknown':
+      return 'r'
+    case 'terran':
+      return 't'
+    case 'zerg':
+      return 'z'
+    default:
+      return assertUnreachable(name)
+  }
+}
+
+// TODO(2Pac): Create this enum as a subset from an enum of all possible game types that `jssuh`
+// should expose, so we don't have to use these magic numbers here.
+enum SupportedReplayGameType {
```

Comment:
Yeah that's weird, must be pulled from some bnet constants that were reversed. We need more of them than you have here though, BW supports all the `Team` types, CTF, Greed, Slaughter, Sudden Death, *probably* Ladder (unsure about that one)

Reply:
@neivv do you have any insights regarding this?

Reply:
I think Ladder was used for ladder games on old blizz-bnet ladder when that was a thing? Custom isn't a thing afaik.

Considering that for now this enum is just used for giving a name for game type, I'd personally just give some default name to game types we're not aware of instead of having enum that assumes that we managed to cover all cases? Or otherwise there probably should be code explicitly rejecting any unknown `header.gameType` after parsing?

Reply:
Currently I'm displaying "Game Type: Unknown" for replays with game types we don't recognize. Seemed like a good-enough-for-now idea :d

Reply:
You aren't though? You call `replayGameTypeToLabel` on the value in the header, unfiltered, provided the header exists, and the calling code assumes (or the typings you created for jssuh?) that the value in the header matches this enum. Since it does not, and there are possible values that are not in this enum, the `assertUnreachable` call can crash at runtime. A crash is not the same as "displaying Unknown"

Reply:
Oh yeah, you're right. For some reason I tested it without a replay header and thought that was that. I made it return `Unknown` now in that function, instead of crashing.

Reply:
Yeah, as a general rule you should never be using `assertUnreachable` on unvalidated user input (and anything read out of a file counts as unvalidated user input), it's for things that are already verified to be within the type system. (I don't really know why TS let you pass a number as this enum without any kind of cast, but maybe I just missed where that cast was being performed)